top notice: Improve top-of-feed notices for search results and combined feed

Yogesh-Shivaji365 opened the pull request:
This PR enhances the top-of-feed notices in Zulip by updating the search results notice and adding a new notice for the combined feed, providing better user experience and clearer navigation options.

Fixes: #33287
<hr>

## Screenshots and screen captures:

**Dark Theme Wrong search.**

| Before | After |
|--------|-------|
| <img width="1919" height="1079" alt="Screenshot 2025-11-02 124545" src="https://github.com/user-attachments/assets/97285fbd-2716-41db-a1b2-07eaa502e443" />|<img width="1919" height="1079" alt="Screenshot 2025-11-07 231643" src="https://github.com/user-attachments/assets/23609a78-742e-4e7f-b127-219c81eac308" />|

**Dark Theme Correct Search.**

| Before | After |
|--------|-------|
|<img width="1919" height="1079" alt="Screenshot 2025-11-02 124603" src="https://github.com/user-attachments/assets/71f61a02-d2d1-40e1-94e6-278a54197247" />| <img width="1919" height="1079" alt="Screenshot 2025-11-02 124442" src="https://github.com/user-attachments/assets/272231e1-9339-494e-ab6d-adcde8e67303" />|

**Dark Theme Combined feed.**

| Before | After |
|--------|-------|
|<img width="1919" height="1079" alt="Screenshot 2025-11-02 124635" src="https://github.com/user-attachments/assets/f22c89be-fb1b-4777-a72e-3345c5afcdbc" />|<img width="1902" height="1072" alt="Screenshot 2025-11-02 124336" src="https://github.com/user-attachments/assets/a9c04641-f806-48df-bd16-9399161ac0ed" />|

**Light Theme Wrong Search.**

| Before | After |
|--------|-------|
|<img width="1919" height="1079" alt="Screenshot 2025-11-02 124537" src="https://github.com/user-attachments/assets/eb4d70e3-139f-4427-89a4-6dd4f9191c4e" />|<img width="1919" height="1079" alt="Screenshot 2025-11-07 231919" src="https://github.com/user-attachments/assets/7f53f229-84c2-4844-b5a3-08763c8f358f" />|

**Light Theme Correct Search.**

| Before | After |
|--------|-------|
|<img width="1919" height="1078" alt="Screenshot 2025-11-02 124612" src="https://github.com/user-attachments/assets/1007ffb7-5043-4cb8-9e75-35b0aba3541e" />|<img width="1900" height="1072" alt="Screenshot 2025-11-02 124457" src="https://github.com/user-attachments/assets/eaf76a27-45db-4d9d-b3f5-847786b12914" />|

**Light Theme Combined feed.**

| Before | After |
|--------|-------|
|<img width="1919" height="1079" alt="Screenshot 2025-11-02 124627" src="https://github.com/user-attachments/assets/d1c443aa-360d-4350-9bf3-9519b8e5a6d4" />|<img width="1919" height="1079" alt="Screenshot 2025-11-02 124326" src="https://github.com/user-attachments/assets/42c5980f-14fc-4810-95ab-2b66f755126e" />|

<details>
<summary>Self-review checklist</summary>

- [X] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
@Yogesh-Shivaji365 Please remove extraneous AI-generated info from your PR description. There's no need to duplicate information that's obvious from looking at the diff.

laurynmm commented:
Please read the documentation that I linked to and follow those guidelines on writing the commit messages for your changes.

Yogesh-Shivaji365 committed:
search: Remove icon and outline from search notice.

Removes the magnifying glass icon. This includes:

- Removing the icon element from the template
- Removing border and box-shadow styles from the notice
- Updating both light and dark theme styles.

Yogesh-Shivaji365 committed:
search: Add search all public channels button.

Adds a button to the end-of-search-results notice that allows users to search all public channels. The button appears in the correct search case.Includes basic click handler with navigation to public channels
search.

Yogesh-Shivaji365 committed:
search: Implement no search results behavior.

Adds a button to the empty feed notice for wrong search cases and ensures proper notice visibility. Uses minimal MutationObserver to hide original notice in wrong search cases while preserving correct search behavior.

Includes click handler for wrong search button with proper
navigation to public channels search and updates tests to
reflect the new template structure with proper indentation.

Yogesh-Shivaji365 committed:
search: impliment combined feed notice in combined feed page.

Replace manual hash checking with `narrow_state.filter()?.is_in_home()`
which is the standard approach used elsewhere in the codebase for
detecting combined feed view or in combined feed view while hiding it in correct/wrong search cases.

Fixes: #33287

laurynmm reviewed (commented):
@Yogesh-Shivaji365 - Thanks for the ping! Your commit messages look better, but some of them still do not follow our [formatting guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#formatting-guidelines).

I had some comments and questions about your updated changes. Let me know here or on CZO if you have any questions.

Yogesh-Shivaji365 commented on web/templates/empty_feed_notice.hbs:
```diff
@@ -1,21 +1,29 @@
+            <button class="empty-search-shared-history action-button action-button-quiet-neutral" type="button">
+                Search all public channels
+            </button>
+        </div>
```
@laurynmm The issue discussion in[ CZO](https://chat.zulip.org/#narrow/channel/101-design/topic/top-of-feed.20notices.20.2333287/near/2071515) indicates that for no results notices, we should have the button directly after the message without extra text above it. The current placement follows this approach.

laurynmm commented on web/src/message_feed_top_notices.ts:
```diff
@@ -31,9 +30,10 @@ function show_end_of_results_notice(): void {
+    $(".all-messages-search-caution a.search-shared-history").attr(
+        "href",
+        "help/search-for-messages#search-shared-history",
+    );
```
There's no need to set this here instead of the template. You do need to update this function for the removal of the shared history link in the template.

laurynmm commented on web/templates/message_feed_errors.hbs:
```diff
@@ -12,25 +12,21 @@
-                    Consider <z-link>searching all public channels that you can view</z-link>.
-                    {{#*inline "z-link"}}<a class="search-shared-history" href="">{{> @partial-block}}</a>{{/inline}}
-                {{/tr}}
+                Search all public channels that you can view
```
Hmmm... does the issue or any of the design discussions note what the button text should be for guests? This seems a bit long.

laurynmm reviewed (commented):
@Yogesh-Shivaji365 - I've done another review round for your changes. Some of my feedback from the previous round wasn't implemented as I intended, so let me know if anything is still unclear.

Also, please make sure that your commit messages follow our guidelines about formatting. I've already linked to that documentation many times, so you can find it in a comment above in this PR thread.

Yogesh-Shivaji365 committed:
Merge branch 'main' into top-notice-fixes

Yogesh-Shivaji365 commented:
Closing this PR because #34814 got merged.
